### ISSUE TITLE
Impossible to have rtd-records with 7 fields

### ISSUE THREAD

**Issue by smoes** [active-group/active-clojure#31]
It is impossible to have rtd-records with 7 fields (at least in cljs):

```
(rec/define-record-type Foo
  {:rtd-record? true}
  (make-foo a b c d e f g) foo?
  [a foo-a
   b foo-b
   c foo-c
   d foo-d
   e foo-e
   f foo-f
   g foo-g])

(make-foo 1 2 3 4 5 6 7)
=> Foo{:a 1, :b 2, :c 3, :d 4, :e 5, :f 6}
```

It is however possible with 6 or 8 fields.

```
(rec/define-record-type Bar
  {:rtd-record? true}
  (make-bar a b c d e f g h) bar?
  [a bar-a
   b bar-b
   c bar-c
   d bar-d
   e bar-e
   f bar-f
   g bar-g
   h bar-h])

(make-bar 1 2 3 4 5 6 7 8)
=> Bar{:a 1, :b 2, :c 3, :d 4, :e 5, :f 6, :g 7, :h 8}
```

Seems like clear case of number discrimination, bzzz awkward!

**Comment by smoes** [comment]
https://github.com/active-group/active-clojure/blob/master/src/active/clojure/record_runtime.cljc#L108

Seems like this line is buggy.

**Comment by smoes** [comment]
Will be fixed in 0.35
